src/composables/info/feature-info.composable.ts

```diff
@@ -0,0 +1,413 @@
+import useMap, {
+  PROJECTION_LUX,
+  PROJECTION_WEBMERCATOR,
+} from '@/composables/map/map.composable'
+import { listen } from 'ol/events'
+import { featureInfoLayerService } from '../../services/info/feature-info-layer.service'
+import { useFeatureInfoStore } from '@/stores/feature-info.store'
+import {
+  FeatureInfoJSON,
+  FeatureJSON,
+} from '../../services/info/feature-info.model'
+import { MapBrowserEvent } from 'ol'
+import { ref, watchEffect } from 'vue'
+import { storeToRefs } from 'pinia'
+import { useAppStore } from '@/stores/app.store'
+import { screenSizeIsAtLeast } from '@/services/common/device.utils'
+import useThemes from '@/composables/themes/themes.composable'
+import { transform } from 'ol/proj'
+import { useDrawStore } from '@/stores/draw.store'
+import { FeatureLike } from 'ol/Feature'
+import { DrawnFeature } from '@/services/draw/drawn-feature'
+import { Pixel } from 'ol/pixel'
+
+const INFO_SERVICE_URL = import.meta.env.VITE_GET_INFO_SERVICE_URL
+
+export default function useFeatureInfo() {
+  const map = useMap().getOlMap()
+  const { setFeatureInfoPanelContent, setLoading, setInfoPanelHidden } =
+    useFeatureInfoStore()
+  const { fid, isLoading } = storeToRefs(useFeatureInfoStore())
+  const { toggleInfoOpen } = useAppStore()
+  const { layersOpen, myMapsOpen } = storeToRefs(useAppStore())
+  const { findById } = useThemes()
+  const { drawStateActive, editStateActive, drawnFeatures } = storeToRefs(
+    useDrawStore()
+  )
+  const responses = ref<FeatureInfoJSON[]>([])
+  const lastHighlightedFeatures = ref<FeatureJSON[]>([])
+  const isLongPress = ref(false)
+  const startPixel = ref<number[] | null>([])
+  const stopPixel = ref<number[] | null>([])
+  const pointerDownTime = ref(0)
+  const pointerUpTime = ref(0)
+  const timeoutId = ref<number | null>(null)
+
+  function init() {
+    featureInfoLayerService.init(map)
+    listen(map, 'pointerdown', event => {
+      ;(() => {
+        isLongPress.value = false
+        startPixel.value = (event as MapBrowserEvent<any>).pixel
+        pointerDownTime.value = new Date().getTime()
+        if (timeoutId.value !== null) {
+          clearTimeout(timeoutId.value)
+        }
+      })()
+    })
+
+    listen(map, 'pointerup', event => {
+      ;(async () => {
+        const evt = event as MapBrowserEvent<any>
+        if (timeoutId.value !== null) {
+          clearTimeout(timeoutId.value)
+          timeoutId.value = null
+        }
+
+        const tempTime = new Date().getTime()
+        if (tempTime - pointerUpTime.value <= 499) {
```

Comment:
I think this one is throttle? you can find one in the `utils.ts`

Reply:
Alright, I've refactored the composable to make use of it.